### PR TITLE
refactoring get_nam.pl into status and data components

### LINKED ISSUE #718: separate status and data acquisition into separate steps for meteorology data
The `get_nam.pl` and `get_atcf.pl` scripts combine status monitoring of meteorological data feeds with downloading of new data (if it is found). In order to achieve greater flexibility and maintainability, the `get_nam.pl` script should refactored to separate these tasks (refactoring the `get_atcf.pl` script will be left for later). 

In order to avoid interruptions in Operations, the `get_nam.pl` will be left in place, and two new scripts will be created to implement the same functionality: `get_nam_status.pl` and `get_nam_data.pl`.

### PR DESCRIPTION
**Pull request by jasonfleming**
resolves #718

### CODE REVIEW

File: NAMtoOWIRamp.pl [hunk 1]
```diff
@@ -535,7 +390,7 @@ sub rotateAndFormat {
 ################################################################################
 sub addToFort22 {
     open( F22, ">>$fort22" )
-      || die "ERROR: NAMtoOWIRamp.pl: Failed to open OWI (NWS12) fort.22 file to append a comment line with the met time increment.";
+      || die "ERROR: $this: Failed to open OWI (NWS12) fort.22 file to append a comment line with the met time increment.";
```
Comment (wwlwpd) [inline NAMtoOWIRamp.pl:393]: You can use `$0` here

File: asgs_main.sh [hunk 2]
```diff
@@ -957,22 +1001,14 @@ downloadBackgroundMet()
...
-   cp -f $STATEFILE ${STATEFILE}.old 2>> ${SYSLOG} 2>&1
-   sed 's/ADVISORY=.*/ADVISORY='$newAdvisoryNum'/' $STATEFILE > ${STATEFILE}.new
-   logMessage "Updating statefile $STATEFILE with new cycle number ${newAdvisoryNum}." $APPLOGFILE
-   cp -f ${STATEFILE}.new $STATEFILE 2>> ${SYSLOG} 2>&1
-   # write the start and end dates of the forecast to the run.properties file
-   if [[ -e $RUNDIR/forecast.properties ]]; then
-      cat $RUNDIR/forecast.properties >> ${SCENARIODIR}/run.properties
-      mv $RUNDIR/forecast.properties ${SCENARIODIR} 2>> ${SYSLOG}
+   # download nowcast data
+   else
```
Comment (wwlwpd) [inline asgs_main.sh:1010]: I guess you have plans for this `else`?

File: get_nam_data.pl [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+my $stage = "null";       # nowcast | forecast
+our $forecastcycle = "00,06,12,18";   # nam cycles to run a forecast
+my $backsite = "ftp.ncep.noaa.gov";   # ncep ftp site for nam data
+my $backdir = "/pub/data/nccf/com/nam/prod";    # dir on ncep ftp site
+my $localbasedir = cwd();    # main directory to download NAM data to
+our $scriptdir = dirname(__FILE__); # directory where this script is installed
+our $forecastlength = 84; # standard length of a NAM forecast
+our $max_retries = 20; # max number of times to attempt download of forecast file
+our $num_retries = 0;
+our $had_enough = 0;
```
Comment (wwlwpd) [inline get_nam_data.pl:72]: this is not a package so you can change the `our` to `my`

**Review by wwlwpd — COMMENTED** [review]
so far so good, let me know when you have the commit ready for merge - proof's in the pudding as far as I'm concerned

File: NAMtoOWIRamp.pl (hunk 1 above)
Comment (jasonfleming) [reply NAMtoOWIRamp.pl]: Thank you, I have made those changes as suggested.

File: get_nam_data.pl (hunk 3 above)
Comment (jasonfleming) [reply get_nam_data.pl]: I have removed all the "our" declarations. Not a big deal on `get_nam_data.pl`, but a lot more of an issue with `NAMtoOWIRamp.pl`. But we made it nonetheless.

File: asgs_main.sh (hunk 2 above)
Comment (jasonfleming) [reply asgs_main.sh]: Yes :-) That was only half-finished at the time, but now it is finished.

**Comment by jasonfleming** [comment]
@wwlwpd This is ready for you to have a look. I have successfully tested it on my desktop machine, running the ec95d mesh with NAM nowcasts and forecasts. So I think it is ready to move from development to staging, which would involve squashing and rebasing to the latest master. Then I'd like to test it again, to make sure nothing got broken in that process. Also, I'd like to run it for a couple days on a real HPC with one of our production meshes. Once that works ok, I think it is safe to merge to master.

Finally, the `namBlend` option is not included in these changes, and I will need to tackle that in the next sprint, rather than prolong this cycle any longer. Let me know what you think. :-)

**Review by wwlwpd — CHANGES_REQUESTED** [review]
@jasonfleming - please place any new .pl scripts you added into `./bin`. Don't worry about ones that already were in `$SCRIPTDIR`. I should take care of that at some point, though.

This directory is in `PATH`, so they can be called without full or relative paths as long as they're set to executable.

**Comment by jasonfleming** [comment]
Hey @wwlwpd I have made these changes and am testing them on my desktop machine. All good so far. Please let me know if any other changes are required.

**Comment by wwlwpd** [comment]
replacing with squash, #755

**Comment by jasonfleming** [comment]
A lot of these commits are not present in current master.

**Pull request closed without merging**